app/services/update_type_service.rb

```diff
@@ -42,19 +42,36 @@ def call(params)
 
   def set_params_and_validate(params)
     # Set patterns includes a data validation before assigning the value to the attribute.
-    # A failure should return a failure.
-    set_patterns(params) if params[:patterns].present?
-    return [false, type.errors] if type.errors.any?
+    # A validation failure should return a service call failure.
+    if params[:patterns].present?
+      validate_enterprise_action(params[:patterns])
+      set_patterns(params[:patterns])
+      return [false, type.errors] if type.errors.any?
+    end
 
     super
   end
 
-  def set_patterns(params)
+  def validate_enterprise_action(patterns)
+    change_from_manual_to_generated = !type.patterns.subject.enabled? && patterns.dig(:subject, :enabled) == true
```

Comment:
:green_circle: Is the `== true` necessary?

```suggestion
    change_from_manual_to_generated = !type.patterns.subject.enabled? && patterns.dig(:subject, :enabled)
```

Reply:
Now, that is an old question :D In a typed language, I'd say "No". But here you could have any value at key `enabled`.

Generally I agree with you though